2760 (#27)
lightningdevkit/ldk-review-club: _posts/2023-12-06-#2760.md

---
layout: pr
date: 2023-00-00
title: "Fix infinite loop when closing a pre-funding channel"
pr: 2760
authors: [jbesraa]
components: ["channel-establishment", "channel-funding"]
host: jbesraa
status: upcoming
commit:
---

## Notes
  Creating a channel involves exchanging multiple messages with a
  connected peer, which does not always end up successfully.

>   Specificly, in the current implemntation an infinite loop might occur 

Reviewer: ```suggestion
  Specifically, in the current implementation an infinite loop might occur 
```

>   in the funding step.

Reviewer: I think we should mention that this bug resulted from https://github.com/lightningdevkit/rust-lightning/pull/2077 specifically.

  PR #2760 Addresses this issue.

        +-------+                         +-------+
        |       |--(1)- open_channel ---->|       |

>         |       |<-(2)- accep_channel ----|       |

Reviewer: ```suggestion
        |       |<-(2)- accept_channel ----|       |
```

        |       |--(3)- funding_created ->|       |
        |       |<-(4)- funding_signed ---|       |
        |       |--(5)- channel_ready --->|       |
        |   A   |<-(6)- channel_ready  ---|   B   |
        +-------+                         +-------+

* Channel establishment(v1) is the process of creating a channel with a
  connected peer, as detailed here  [lightning/bolt02].

## Questions
1. Did you review the PR? [Concept ACK, approach ACK, tested ACK, or NACK](https://github.com/lightningdevkit/rust-lightning/blob/master/CONTRIBUTING.md#peer-review)?
2. In the original code we have a loop inside `close_channel_internal`, 
in which scenario it goes indefintly? 

> 3. why are we able to drop `force_close_channel_with_peer` from

Reviewer: ```suggestion
3. Why are we able to drop `force_close_channel_with_peer` from
```

`close_channel_internal`, and `funding_created` msg from `SignerResumeUpdate`?

> 4. question about v2 channel establishemnt
> 5. question about v2 channel establishemnt

Reviewer: ```suggestion
4. What was the reasoning for promoting an `OutboundV1Channel` to a `Channel` before receiving `funding_signed` prior to this PR? Why do we no only do that promotion after receiving `funding_signed`?
```

---
layout: pr

> date: 2023-00-00

Reviewer: We still need a valid date here otherwise it won't compile :)
